plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/StmtConversionVisitor.kt

```diff
@@ -298,14 +312,49 @@ object StmtConversionVisitor : FirVisitor<Exp, StmtConversionContext<ResultTrack
         variableAssignment: FirVariableAssignment,
         data: StmtConversionContext<ResultTrackingContext>,
     ): Exp {
+
         // It is not entirely clear whether we can get away with ignoring the distinction between
         // lvalues and rvalues, but let's try to at first, and we'll fix it later if it turns out
         // not to work.
-        val convertedLValue = data.convert(variableAssignment.lValue)
         val lValueType = data.embedType(variableAssignment.lValue)
         val convertedRValue = data.convert(variableAssignment.rValue)
         val rValueType = data.embedType(variableAssignment.rValue)
-        data.addStatement(Stmt.assign(convertedLValue, convertedRValue.convertType(rValueType, lValueType)))
+
+        if (variableAssignment.lValue.isAssigningToClassField) {
+
+            val lValue = variableAssignment.lValue as FirPropertyAccessExpression
+            val lValueSymbol = lValue.calleeSymbol as FirPropertySymbol
+            val receiver = data.convert(lValue.dispatchReceiver!!)
+            /*
+            According to the setter symbol, we should invoke a method, or assigning a value.
+
+            Setters are only relevant for variable properties.
+            Compared to the getter's, the setter symbols require more information about the backing field.
+            If a custom setter is defined on a variable property. this also exposes a backing field.
+            To check the existence of a custom setter, we can look for a function body in a `setterSymbol`.
+            */
+            val setterSymbol = lValueSymbol.setterSymbol!!
+            if (setterSymbol.hasBody) {
+                // Since a custom setter has been defined, we should generate a statement to invoke a method call
+                val method = data.embedFunction(setterSymbol)
+                data.withResult(UnitTypeEmbedding) {
+                    data.addStatement(method.toMethodCall(listOf(receiver, convertedRValue), this.resultCtx.resultVar))
+                }
+            } else {
+                // No custom setters have been defined, we can assign the fields normally.
+                val varEmbedding = VariableEmbedding(lValueSymbol.callableId.embedName(), lValueType)
```

Comment:
I think this should be constructed via the context now.  Could you add a test where a property is used in an inline function?